airflow/jobs/scheduler_job.py

```diff
@@ -567,302 +567,71 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
-                            'dagrun.schedule_delay.{dag_id}'.format(dag_id=dag.dag_id),
-                            schedule_delay)
-                    self.log.info("Created %s", dag_run)
-
-            if dag_runs_for_dag:
-                tis_out.extend(self._process_task_instances(dag, dag_runs_for_dag))
-                if check_slas:
-                    self.manage_slas(dag)
-
-        return tis_out
-
-    def _find_dags_to_process(self, dags: List[DAG]) -> List[DAG]:
-        """
-        Find the DAGs that are not paused to process.
-
-        :param dags: specified DAGs
-        :return: DAGs to process
-        """
-        if self.dag_ids:
-            dags = [dag for dag in dags
-                    if dag.dag_id in self.dag_ids]
-        return dags
-
-    @provide_session
-    def execute_on_failure_callbacks(
+    def execute_callbacks(
         self,
         dagbag: DagBag,
-        failure_callback_requests: List[FailureCallbackRequest],
+        callback_requests: List[CallbackRequest],
         session: Session = None
     ) -> None:
         """
         Execute on failure callbacks. These objects can come from SchedulerJob or from
         DagFileProcessorManager.
 
-        :param failure_callback_requests: failure callbacks to execute
-        :type failure_callback_requests: List[airflow.utils.dag_processing.FailureCallbackRequest]
+        :param dagbag: Dag Bag of dags
+        :param callback_requests: failure callbacks to execute
+        :type callback_requests: List[airflow.utils.callback_requests.CallbackRequest]
         :param session: DB session.
         """
-        for request in failure_callback_requests:
-            simple_ti = request.simple_task_instance
-            if simple_ti.dag_id in dagbag.dags:
-                dag = dagbag.dags[simple_ti.dag_id]
-                if simple_ti.task_id in dag.task_ids:
-                    task = dag.get_task(simple_ti.task_id)
-                    ti = TI(task, simple_ti.execution_date)
-                    # Get properties needed for failure handling from SimpleTaskInstance.
-                    ti.start_date = simple_ti.start_date
-                    ti.end_date = simple_ti.end_date
-                    ti.try_number = simple_ti.try_number
-                    ti.state = simple_ti.state
-                    ti.test_mode = self.UNIT_TEST_MODE
+        for request in callback_requests:
+            try:
+                if isinstance(request, TaskCallbackRequest):
+                    self._execute_task_callbacks(dagbag, request)
+                elif isinstance(request, SlaCallbackRequest):
+                    self.manage_slas(dagbag.dags.get(request.dag_id))
+                elif isinstance(request, DagCallbackRequest):
+                    self._execute_dag_callbacks(dagbag, request, session)
+            except Exception:  # pylint: disable=broad-except
+                self.log.exception("Error executing callback for File: %s", request.full_filepath)
```

Comment:
```suggestion
                self.log.exception("Error executing %s callback for file: %s", request.__class__.__name__, request.full_filepath)
```
Just to give more information

Reply:
Will do